File: part-1/set-2/cpp/challenge-1/Employee.cpp
#include "Employee.h"

// Constructors
Employee::Employee() : name(""), age(0), department("") {}

Employee::Employee(const std::string &name, int age, const std::string &department)
    : name(name), age(age), department(department) {}

Employee::Employee(const std::string &department, const std::string &name, int age)
    : name(name), age(age), department(department) {}

std::string Employee::getName() const
{
    return name;
}

// Get and set
int Employee::getAge() const
{
    return age;
}

std::string Employee::getDepartment() const
{
    return department;
}

void Employee::setName(const std::string &newName)
{
    name = newName;
}

void Employee::setAge(int newAge)
{
    age = newAge;
}

void Employee::setDepartment(const std::string &newDepartment)
{
    department = newDepartment;
}

File: part-1/set-2/java/challenge-1/Employee.java
// Program: Employee Management (Java)
// Description: Demonstrates an Employee class with private variables, constructor, getters, and setters

public class Employee {
    private String department;
    private String name;
    private int age;

    public Employee(){
        this.department = "";
        this.age = 0;
        this.name = "";
    }

    public Employee(String department, String name, int age){
        this.department = department;
        this.name = name;
        this.age = age;
    }

    public Employee(String name, int age, String department){
        this.department = department;
        this.name = name;
        this.age = age;
    }

    public String getName(){
        return this.name;
    }

    public void setName(String name){
        this.name = name;
    }

    public String getDepartment(){
        return this.department;
    }

    public void setDepartment(String department){
        this.department = department;
    }

    public int getAge(){
        return this.age;
    }

    public void setAge(int age){
        this.age = age;
    }
}
